opencog/atoms/core/Variables.cc

```diff
@@ -238,43 +250,41 @@ bool VarScraper::less_than(const Handle& lh, const Handle& rh,
 		// Both atoms have same arity, sort by outgoings
 		const HandleSeq& lout = lh->getOutgoingSet();
 		const HandleSeq& rout = rh->getOutgoingSet();
-		return is_ordered_link(lh)? less_than_ordered(lout, rout, ctx)
-			: less_than_unordered(lout, rout, ctx);
+		return is_ordered_link(lh) ? less_than_ordered_outgoing(lout, rout)
+			: less_than_unordered_outgoing(lout, rout);
 	}
 
-	// Both atoms are constant nodes of the same type, sort by regular
-	// atom order.
+	// None are variables, sort by node content.
 	if (not is_variable(lh))
 		return lh < rh;
 
-	// Both atoms are variables. If they are have different counts,
-	// sort by counts, as that affects their semantics.
-	unsigned lc = _fvc.get(lh);
-	unsigned rc = _fvc.get(rh);
-	if (lc != rc)
-		return lc < rc;
+	const auto& lps = _paths.at(lh);
+	const auto& rps = _paths.at(rh);
+	if (lps != rps)
+		return lps < rps;
 
-	// Both atoms are variables with same count, sort by regular atom
-	// order.
+	// Still a tie? These 2 variables are semantically equivalent. Sort
+	// by atom order then.
 	return lh < rh;
 }
 
-bool VarScraper::less_than_ordered(const HandleSeq& lhs, const HandleSeq& rhs,
-                                   const Context& ctx) const
+bool VarScraper::less_than_ordered_outgoing(const HandleSeq& lhs,
+                                            const HandleSeq& rhs) const
 {
```

Comment:
Now that `const Context& ctx` is gone, this seems to be a confusing duplicate of `handle_seq_less`  in `Handle.h` (Which, BTW, seems mis-declared, I think it should be declared as `namespace std { template <> struct less<opencog::HandleSeq>; }` or something like that ...

Reply:
See

https://github.com/opencog/atomspace/blob/e97da6f8d4d7b5a7ef921d99e11942895f1b208c/opencog/atoms/core/Variables.cc#L261-L264

to understand the fundamental difference.

Reply:
Yes, sure, but I wasn't talking about that function, I was pointing at `less_than_ordered_outgoing()`

Reply:
Ohhh, because its recursively embedded deep in there. OK.